helix-view/src/document.rs

```diff
@@ -605,32 +605,38 @@ impl Document {
             return Some(formatting_future.boxed());
         };
 
-        let language_server = self.language_server()?;
-        let text = self.text.clone();
-        let offset_encoding = language_server.offset_encoding();
-
-        let request = language_server.text_document_formatting(
-            self.identifier(),
-            lsp::FormattingOptions {
-                tab_size: self.tab_width() as u32,
-                insert_spaces: matches!(self.indent_style, IndentStyle::Spaces(_)),
-                ..Default::default()
+        match self.language_server() {
+            Some(language_server) => {
+                let text = self.text.clone();
+                let offset_encoding = language_server.offset_encoding();
+
+                let request = language_server.text_document_formatting(
+                    self.identifier(),
+                    lsp::FormattingOptions {
+                        tab_size: self.tab_width() as u32,
+                        insert_spaces: matches!(self.indent_style, IndentStyle::Spaces(_)),
+                        ..Default::default()
+                    },
+                    None,
+                )?;
+
+                let fut = async move {
+                    let edits = request.await.unwrap_or_else(|e| {
+                        log::warn!("LSP formatting failed: {}", e);
+                        Default::default()
+                    });
+                    Ok(helix_lsp::util::generate_transaction_from_edits(
+                        &text,
+                        edits,
+                        offset_encoding,
+                    ))
+                };
+                    Some(fut.boxed())
             },
-            None,
-        )?;
-
-        let fut = async move {
-            let edits = request.await.unwrap_or_else(|e| {
-                log::warn!("LSP formatting failed: {}", e);
-                Default::default()
-            });
-            Ok(helix_lsp::util::generate_transaction_from_edits(
-                &text,
-                edits,
-                offset_encoding,
-            ))
-        };
-        Some(fut.boxed())
+            None => {
+                Some(async move { return Err(FormatterError::LanguageServerNotInitialized) }.boxed())
+            }
```

Comment:
`Document::language_server` returning None can happen if you don't have the language server installed for a language. We shouldn't show an error if we can't format because the language server isn't installed - it would be too noisy

Reply:
The code needs more work, but that is exaclty what the issue wants resolved. Do we still not want to show errors in those cases (when language server isn't installed)?

Reply:
I think probably we should show an error in the format type-able command if the lsp is not installed instead

Reply:
Showing errors in `:format` is ok but `make_format_callback` is called by `:write` and `:write-all` as well

Reply:
Oh yeah sorry that's what I meant should have been more clear

Reply:
sounds good, thanks